lib/linguist/heuristics.yml

```
@@ -630,6 +630,12 @@ disambiguations:
     pattern: '^%(end|ctor|hook|group)\b'
   - language: Linker Script
     pattern: 'OUTPUT_ARCH\(|OUTPUT_FORMAT\(|SECTIONS'
+- extensions: ['.yaml']
+  rules:
+  - language: MiniYAML
+    pattern: '^\t+.*:'
```

If you intend to match tab-indented fields like `\tfoo:`, then I recommend matching at least one non-whitespace, non-colon character as part of the field-name:

```suggestion
    pattern: '^\t+.*?[^\s:].*?:'
```

---

Curious, what's the difference between `.*` and `.*?`, aren't they the same given that `*` matches empty as well and `?` just means optional?

---

Normally, a regex will attempt to match the longest possible string that it can; for example:

~~~perl
/Foo.+/ ~= "Foo_123456789" # => returns the entire line, "Foo_123456789"
~~~

Affixing `?` to a quantifier enables a so-called "conservative" mode (a.k.a., "thrifty", "cautious", "non-greedy", etc). When marked as thrify, a quantifier will match the fewest characters it needs to in order to satisfy the criteria:

~~~perl
/Foo.+?/ ~= "Foo_123456789" # Returns only "Foo_"
~~~

Thrify expressions can be decidedly more performant than greedy ones, but only in the event of a successful match.

Note that you can't quantify a quantifier; an optional `+` or `?` wouldn't make sense. 😉 Hence why the syntax was chosen to avoid conflicting with other regular expression flavours. You can thank Perl for that.

---

Updated.